service/service.go

```diff
@@ -0,0 +1,53 @@
+// Package implements a function, that generates a random sequence of parentheses with length = n, and handler,
+// that handle request to /generate?n={length}
+package service
+
+import (
+	"errors"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+var ErrNegativeLength = errors.New("negative length")
+
+// HandleRequests implements all handlers
+func HandleRequests() {
+	http.HandleFunc("/generate", GenerateHandler)
+}
+
+// GenerateHandler handles request to /generate?n={length}
+func GenerateHandler(w http.ResponseWriter, r *http.Request) {
+	n := r.URL.Query().Get("n")
+	l, err := strconv.Atoi(n)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+	}
+
+	s, err := ParenthesesGeneration(l)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+	}
```

Comment:
it looks strange that an error in `ParenthesesGeneration` produces a Bad Request.
Check input in your handler right away. Business Logic should produce business errors